my-app/api/user/add.ts

```diff
@@ -0,0 +1,27 @@
+import { User } from "@/interfaces/user";
+import { ApiResponse } from "@/interfaces/api-response";
+import { apiClient } from "@/api";
+import { AxiosError, isAxiosError } from "axios";
+import { PossibleRolesEn } from "@/utils";
+
+interface AddUserRequest {
+  name: string;
+  email: string;
+  phone: string;
+  birthDate: string;
+  password: string;
+  role: PossibleRolesEn;
+}
+
+export async function add(request: AddUserRequest): Promise<ApiResponse<User>> {
```

Comment:
```suggestion
export async function addUser(request: AddUserRequest): Promise<ApiResponse<User>> {
```

Apesar de estar dentro da pasta user, tanto o módulo `add` quando a função `add` podem não ficar muito claros na leitura, se vc considerar a perspectiva de alguém que abre apenas o arquivo em que essa função tá sendo chamada. `addUser` é muito mais direto a leitura, concorda?